electron/src/lib/CoreProtocol.ts

```diff
@@ -0,0 +1,80 @@
+/*
+ * Wire
+ * Copyright (C) 2019 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+import {LogFactory} from '@wireapp/commons';
+import {app} from 'electron';
+import * as path from 'path';
+import {URL} from 'url';
+import {platform} from '../js/environment';
+import {AutomatedSingleSignOn} from './AutomatedSingleSignOn';
+
+const LOG_DIR = path.join(app.getPath('userData'), 'logs');
+const LOG_FILE = path.join(LOG_DIR, 'electron.log');
+const logger = LogFactory.getLogger('CoreProtocol', {forceEnable: true, logFilePath: LOG_FILE});
+
+const {customProtocolName} = require('../../package.json');
```

Comment:
`customProtocolName` seems to be defined in `info.json` and not `package.json`.

Reply:
It is defined in `package.json` after build is done.

Reply:
Why do we need to have 2 places to store that information?

Reply:
One is for build time, the other at runtime

Reply:
Can one be replaced with the other?

Reply:
Thanks for the detailed explanation! Will mark this discussion as resolved. 👍